.circleci/should_publish.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+IFS='£' read -a PUB_TAGS <<< "${PUBLISHABLE_TAGS}"
+for item in ${PUB_TAGS[@]}; do
+    [[ "${CIRCLE_TAG}" =~ ${item} ]] && echo "true" && exit 0
+done
+
+IFS='£' read -a PUB_BRANCHES <<< "${PUBLISHABLE_BRANCHES}"
```

Comment:
@gjedlicska

Reply:
This publishes on _any_ commit to these branches, or _any_ tag (matching the semver regex).

Reply:
haha, good Q. The idea of filtering tags for specific branches is unforch not going to work. If you check out a git tag ( which is what CircleCI does ) the git HEAD is in a detached state, with no connections to its parent branch. This i knew, but forgot, when we were planning this, so sorry about that...

What we can do is have two separate trigger conditions
- publish on tag matching semver stable range
- publish on all commit for branches `main | hotfix.* | alpha.*`

for the branch commits, we rely on the version number properly generated by the get version step